### PR TITLE
ipn,types/persist: add DisallowedTKAStateIDs, refactor as view type

### PR DESCRIPTION
**Pull request by twitchyliquid64**
Supercedes https://github.com/tailscale/tailscale/pull/6557, precursor to trying https://github.com/tailscale/tailscale/pull/6546 again

### CODE REVIEW

File: types/persist/persist_test.go [hunk 1]
```diff
@@ -133,6 +133,11 @@ func TestPersistEqual(t *testing.T) {
 			&Persist{NodeID: "abc"},
 			false,
 		},
+		{
+			&Persist{DisallowedTKAStateIDs: nil},
+			&Persist{DisallowedTKAStateIDs: []string{"0:0"}},
+			false,
+		},
```
Comment (maisem) [inline types/persist/persist_test.go:140]: add the positive case too

**Review by maisem — APPROVED** [review]

File: types/persist/persist.go [hunk 2]
```diff
@@ -70,7 +77,8 @@ func (p *Persist) Equals(p2 *Persist) bool {
 		p.LoginName == p2.LoginName &&
 		p.UserProfile == p2.UserProfile &&
 		p.NetworkLockKey.Equal(p2.NetworkLockKey) &&
-		p.NodeID == p2.NodeID
+		p.NodeID == p2.NodeID &&
+		reflect.DeepEqual(p.DisallowedTKAStateIDs, p2.DisallowedTKAStateIDs)
```
Comment (bradfitz) [inline types/persist/persist.go:81]: do you care about nil vs empty slice? this treats them differently, but I doubt that matters. you might want to use that len-zero-map-to-nil func we have somewhere

Comment (bradfitz) [reply types/persist/persist.go:81]:
```
bradfitz@tsdev:~/src/tailscale.com$ git grep ^func | grep nil | grep any
wgengine/router/ifconfig_windows_test.go:func nilIfEmpty[E any](s []E) []E {
```

Comment (twitchyliquid64) [reply types/persist/persist.go]: Done.

**Pull request merged**